.github/workflows/go.yml

```diff
@@ -22,15 +21,22 @@ jobs:
     - name: Check out code into the Go module directory
       uses: actions/checkout@v2
 
-    - name: Get dependencies
+    - name: Initialize dependencies and linters
+      run: |
+        make init
+
+    - name: Diff
       run: |
-        go get -v -t -d ./...
-        if [ -f Gopkg.toml ]; then
-            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
-            dep ensure
-        fi
+        make format
+        git diff --exit-code
```

Comment:
I think we could move that to the very end to check that tests do not modify repo state too